mlflow/transformers/__init__.py

```diff
@@ -1333,6 +1333,22 @@ def _extract_torch_dtype_if_set(pipeline):
     if torch_dtype := getattr(pipeline, _TORCH_DTYPE_KEY, None):
         return str(torch_dtype)
 
+    # Transformers pipeline doesn't inherit underlying model's dtype, so we have to extract it
+    # from the model.
+    model = pipeline.model
+    if torch_dtype := getattr(model.config, _TORCH_DTYPE_KEY, None):
+        return str(torch_dtype)
+
+    # Some model doesn't populate dtype in its config, then we try extract it from parameters.
+    if hasattr(model, "parameters"):
+        try:
+            first_param = next(model.parameters())
+        except StopIteration:
+            return None
```

Comment:
```suggestion
        first_param = next(model.parameters(), None)
        if first_param is None:
            return None
```

Can we use `next(..., None)` instead of try-catch?